### PR TITLE
Adding support for Chocolatey.

### LINKED ISSUE #95: Create Chocolatey package

### PR DESCRIPTION
**Pull request by phillipsj**
Adding support for Chocolatey.  Fixes #95 . @gep13 @daveaglick can you all review.  I know I need to add the task to the default task.  @daveaglick would you be interested in stream lining your cake file, I think Cake.Recipe would work nicely.

### CODE REVIEW

File: nuspec/chocolatey/Wyam.Install.nuspec [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <docsUrl>https://wyam.io/docs/</docsUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Wyam is a simple to use, highly modular, and extremely configurable static content generator. This is a tools package that contains the Wyam CLI.</description>
+    <summary>Wyam is a simple to use, highly modular, and extremely configurable static content generator. This is a tools package that contains the Wyam CLI.</summary>
+  <releaseNotes>
+All release notes for Wyam can be found on the GitHub site - https://github.com/Wyamio/Wyam/releases
+  </releaseNotes>
+    <tags>Wyam Static StaticContent StaticSite Blog BlogEngine</tags>
+    <dependencies>
+      <dependency id="Chocolatey" version="[0.10.3, 0.11)" />
```
Comment (gep13) [inline nuspec/chocolatey/Wyam.Install.nuspec]: @phillipsj there is no requirement to have this dependency, unless of course you are using a function that only shipped with Chocolatey 0.10.3.  Did you read that this was required somewhere?

File: nuspec/chocolatey/Wyam.Install.nuspec [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    </dependencies>
+  </metadata>
+  <files>
+    <file src="..\..\build\**\bin\*" target="tools"/>
+    <file src="..\..\LICENSE" target="tools\LICENSE"/>
```
Comment (gep13) [inline nuspec/chocolatey/Wyam.Install.nuspec]: @phillipsj personally, I would put the LICENSE and VERIFICATION files at the root of the package.

**Review by gep13 — CHANGES_REQUESTED** [review]
@phillipsj am I seeing this right that the VERIFICATION.TXT file is empty?  If so, that is going to need to change.  Have a look at the contents of the VERIFICATION.TXT for either Cake or Chocolatey GUI to see what is required.

File: nuspec/chocolatey/Wyam.Install.nuspec (hunk 2 above)
Comment (phillipsj) [reply nuspec/chocolatey/Wyam.Install.nuspec]: @gep13  @daveaglick I made the changes requested.  Please review, if everything is good, I will squash the commits.

File: nuspec/chocolatey/Wyam.Install.nuspec [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <summary>Wyam is a simple to use, highly modular, and extremely configurable static content generator. This is a tools package that contains the Wyam CLI.</summary>
+  <releaseNotes>
+All release notes for Wyam can be found on the GitHub site - https://github.com/Wyamio/Wyam/releases
+  </releaseNotes>
+    <tags>Wyam Static StaticContent StaticSite Blog BlogEngine</tags>
+  </metadata>
+  <files>
+    <file src="..\..\build\**\bin\*" target="tools"/>
+    <file src="..\..\LICENSE" target="LICENSE"/>
+    <file src="VERIFICATION.txt" target="tools"/>
```
Comment (gep13) [inline nuspec/chocolatey/Wyam.Install.nuspec]: @phillipsj in the same way that you have moved the LICENSE file to the root of the Chocolatey package, I would also put the VERIFICATION.txt there too.

**Review by gep13 — CHANGES_REQUESTED** [review]

File: build.cake [hunk 4]
```diff
@@ -377,7 +392,27 @@ Task("Publish-Packages")
...
+    .WithCriteria(()=> isLocal)
+    .Does(()=> 
+    {
+        var chocolateyApiKey = EnvironmentVariable("CHOCOLATEY_API_KEY");
+        if (string.IsNullOrEmpty(chocolateyApiKey))
+        {
+            throw new InvalidOperationException("Could not resolve Chocolatey API key.");
+        }
+
+        foreach(var chocoPgk in GetFiles(chocoRoot.Path.FullPath + "/*.nupkg"))
```
Comment (gep13) [inline build.cake]: @phillipsj there is a minor typo here...  `chocoPgk` -> `chocoPkg`

File: build.cake [hunk 5]
```diff
@@ -377,7 +392,27 @@ Task("Publish-Packages")
...
+        {
+            ChocolateyPush(chocoPgk, new ChocolateyPushSettings {
+                ApiKey = chocolateyApiKey,
+                Source = "https://chocolatey.org/api/v2/"
```
Comment (gep13) [inline build.cake]: @phillipsj the recommended URL to push to is actually `https://push.chocolatey.org/`.  This changed recently.

**Review by gep13 — CHANGES_REQUESTED** [review]

File: build.cake [hunk 6]
```diff
@@ -48,6 +51,7 @@ var semVersion = version + (isLocal ? string.Empty : string.Concat("-build-", bu
 var buildDir = Directory("./src/clients/Wyam/bin") + Directory(configuration);
 var buildResultDir = Directory("./build") + Directory(semVersion);
 var nugetRoot = buildResultDir + Directory("nuget");
+var chocoRoot = buildResultsDir + Directory("choco");
```
Comment (gep13) [inline build.cake]: @phillipsj the AppVeyor build is currently failing.  I think the issue is here... `buildResultsDir` -> `buildResultDir`

**Review by gep13 — CHANGES_REQUESTED** [review]

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@daveaglick did you have any questions about this?

**Comment by daveaglick** [comment]
Nope, just haven't had any OSS time recently. Thanks both for getting this polished - I'll get it pulled in for the next release.

**Comment by daveaglick** [comment]
WOOT! Thanks a ton @phillipsj for doing the legwork on this, and thanks @gep13 for reviewing everything. Hoping to get a release out today or next week so the choco package will get published with that.

**Pull request merged**